samcli/lib/deploy/deployer.py

```diff
@@ -307,8 +307,13 @@ def wait_for_changeset(self, changeset_id, stack_name):
 
         # Wait for changeset to be created
         waiter = self._client.get_waiter("change_set_create_complete")
-        # Poll every 5 seconds. Changeset creation should be fast
+        # If there is a client sleep, use it as the delay or use the default 5 seconds
         waiter_config = {"Delay": 5}
+        if ( 
+            self.client_sleep 
+            and self.client_sleep > waiter_config["Delay"]
```

Comment:
Curious on why you decided to default to it this way. `client_sleep` is set by 
```
try:
  self.client_sleep = float(client_sleep)
except ValueError:
  self.client_sleep = DEFAULT_CLIENT_SLEEP
if self.client_sleep <= 0:
  self.client_sleep = DEFAULT_CLIENT_SLEEP
```

So why not just set it to `self.client_sleep` here?

Reply:
Thanks @jfuss for the reviewing it quickly. I'm basically reading the `self.client_sleep` whatever the value that's getting it into `wait_for_changeset` and verifying with the previously set waiter_config delay value ( 5 seconds ) i think 5 seconds is reasonable for polling and if we use `DEFAULT_CLIENT_SLEEP` of `0.5` it may be too short and might exhaust max attempts quickly. Does this makes sense to you?

Reply:
Sounds reasonable but I think we should just default to `self.client_sleep`. Even if the customer set this to shorter, we should respect that. It can be confusing to someone, if we put a lower limit here. It's likely people will only ever increase but I would rather not lock them into a "not smaller than" time period.

Reply:
Makes sense to me. Just updated to use `self.client_sleep` by default.